pwnlib/util/cyclic.py

```diff
@@ -289,12 +280,8 @@ def cyclic_metasploit(length = None, sets = None):
     sets = sets or [ string.ascii_uppercase.encode(), string.ascii_lowercase.encode(), string.digits.encode() ]
     out = bytearray()
 
-    for ndx, c in enumerate(metasploit_pattern(sets)):
-        if length != None and ndx >= length:
-            break
-        else:
-            out.append(c)
-
+    generator = metasploit_pattern(sets)
+    out = [next(generator) for _ in range(length)] if length != None else list(generator)
     out = bytes(out)
```

Comment:
I suppose this could fix the build failure:
```suggestion
     out = bytes(bytearray(out))
```
or:
```suggestion
     out = _join_sequence(out, sets[0])
```

Reply:
Agree. Will use the latter version.